gvr-arcore/app/src/main/java/org/gearvrf/arcore/simplesample/SampleMain.java

```diff
@@ -0,0 +1,309 @@
+    private static float AR2VR_SCALE = 100;
+
+    private ARCoreHelper mARCoreHelper;
+    private Vector3f mDisplayGeometry;
+
+    private GVRScene mVRScene;
+    private Session mARCoreSession;
+    private GVRSceneObject mARPassThroughObject;
+    private Frame mLastARFrame;
+    private ARCoreHandler mARCoreHandler;
+    private GVRSceneObject mCursor;
+    private GVRCursorController mCursorController;
+
+    /* From AR to GVR space matrices */
+    private float[] mGVRModelMatrix = new float[16];
+    private float[] mARViewMatrix = new float[16];
+    private float[] mGVRCamMatrix = new float[16];
+    private float[] mModelViewMatrix = new float[16];
+
+    public SampleMain() {}
+
+    @Override
+    public void onInit(final GVRContext gvrContext) throws Throwable {
+        super.onInit(gvrContext);
+        if (mARCoreSession == null) {
+            Log.e(TAG, "Invalid ARCore session!");
+            return;
+        }
+
+        mVRScene = gvrContext.getMainScene();
+
+        mARCoreHelper = new ARCoreHelper(gvrContext, mVRScene);
+
+        gvrContext.runOnGlThread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    onInitARCoreSession(gvrContext);
+                     /* Cursor controller */
+                    initCursorController(gvrContext);
+                } catch (CameraNotAvailableException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+    }
+
+    private void onInitARCoreSession(GVRContext gvrContext) throws CameraNotAvailableException {
+        GVRTexture passThroughTexture = new GVRExternalTexture(gvrContext);
+
+        mARCoreSession.setCameraTextureName(passThroughTexture.getId());
+
+        // FIXME: detect VR screen aspect ratio. Using empirical 16:9 aspect ratio
+        /* Try other aspect ration whether virtual objects looks jumping ou sliding
+        during camera's rotation.
+         */
+        mARCoreSession.setDisplayGeometry(Surface.ROTATION_90 , 160, 90);
+
+        mLastARFrame = mARCoreSession.update();
+        mDisplayGeometry = configDisplayGeometry(mLastARFrame.getCamera());
+
+        mARCoreSession.setDisplayGeometry(Surface.ROTATION_90 ,
+                    (int)mDisplayGeometry.x, (int)mDisplayGeometry.y);
+
+        /* To render texture from phone's camera */
+        mARPassThroughObject = new GVRSceneObject(gvrContext, mDisplayGeometry.x, mDisplayGeometry.y,
+                passThroughTexture, GVRMaterial.GVRShaderType.OES.ID);
+
+        mARPassThroughObject.getRenderData().setRenderingOrder(GVRRenderData.GVRRenderingOrder.BACKGROUND);
+        mARPassThroughObject.getRenderData().setDepthTest(false);
+        mARPassThroughObject.getTransform().setPosition(0, 0, mDisplayGeometry.z);
+        mARPassThroughObject.attachComponent(new GVRMeshCollider(gvrContext, true));
+
+        mVRScene.addSceneObject(mARPassThroughObject);
```

Comment:
Because I am updating the pass-through object at frame ratio of phone's cam to avoid frozen frame during the rotation of GVRf's cam. Phone's cam records at 30fps and GVRf renders at 60fps. When we have ARCore working at 60fps  we can attach the  pass-through object to the camera rig.

Reply:
Ok. Btw I tried it and couldn't tell any difference; the camera quad wouldn't "wobble" as much too.